platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookInitializer.java

```diff
@@ -127,7 +127,21 @@ public AddressBookInitializer(
         final AddressBookConfig addressBookConfig =
                 platformContext.getConfiguration().getConfigData(AddressBookConfig.class);
         this.initialState = Objects.requireNonNull(initialState, "The initialState must not be null.");
-        this.stateAddressBook = initialState.isGenesisState() ? null : initialState.getAddressBook();
+
+        // The state address book may be null in some genesis scenarios.  It will be updated to not be null
+        // before it is used in the rest of the platform.  Instead of making the rest of the platform adapt to a
+        // possibly null address book in the API of the state, the null pointer exception is caught here.
+        AddressBook stateAddressBook;
+        try {
+            stateAddressBook = initialState.getAddressBook();
+        } catch (final NullPointerException ex) {
+            if (!initialState.isGenesisState()) {
+                throw new IllegalStateException("Null state address books are only allowed at genesis.");
+            }
+            stateAddressBook = null;
+        }
+        this.stateAddressBook = stateAddressBook;
```

Comment:
Instead of just catching nullpointer exceptions, a better approach I think would be to drill down into the object and directly check. IMO, it's riskly doing this type of check, since it's plausible that we may accidentally catch a different problem that also throws a nullpointer exception.

Reply:
fair point.